### PR TITLE
pause/resume AKPlayer.

### PR DESCRIPTION
**Pull request by ryanfrancesconi**
pause()
resume()
added to AKPlayer. Changes to Loop behavior in regard to pausing. It's now possible to pause and resume a looping player. This has been asked for a bunch of times...

Issue:
https://github.com/AudioKit/AudioKit/issues/1272

### CODE REVIEW

**Comment by aure** [comment]
This PR isn't passing Travis:


❌  /Users/travis/build/AudioKit/AudioKit/AudioKit/Common/Internals/AudioUnit Host/AKAudioUnitManager.swift:140:16: value of type '[AVAudioUnit?]' has no member 'compactMap'
        return _effectsChain.compactMap { $0 }

Are you using Xcode beta?

File: AudioKit/Common/Internals/AudioUnit Host/AKAudioUnitManager.swift [hunk 1]
```diff
@@ -137,7 +137,7 @@ open class AKAudioUnitManager: NSObject {
 
     // just get a non nil list of Audio Units
     private var linkedEffects: [AVAudioUnit] {
-        return _effectsChain.flatMap { $0 }
+        return _effectsChain.compactMap { $0 }
```
Comment (aure) [inline AudioKit/Common/Internals/AudioUnit Host/AKAudioUnitManager.swift]: I think this is too bleeding edge.

**Comment by ryanfrancesconi** [comment]
Oh yeah. The beta.

**Comment by ryanfrancesconi** [comment]
I’ll change it to the old syntax

**Comment by ryanfrancesconi** [comment]
@aure 
image

File: AudioKit/Common/Internals/AudioUnit Host/AKAudioUnitManager.swift (hunk 1 above)
Comment (ryanfrancesconi) [reply AudioKit/Common/Internals/AudioUnit Host/AKAudioUnitManager.swift]: it's just the current syntax, but ok

Comment (ryanfrancesconi) [reply AudioKit/Common/Internals/AudioUnit Host/AKAudioUnitManager.swift]: hard to keep track with swift as they push you to upgrade all the time. I just try to stay on top of the changes now so I don't have all this refactoring to do.

**Comment by megastep** [comment]
We could probably bracket these with `#if swift(4.1)` blocks so it works with both versions.

**Comment by ryanfrancesconi** [comment]
Well, it’s just a deprecation warning. I changed it back to the old syntax for now.
Seems a bit verbose to have to do that?

**Pull request merged**